### PR TITLE
feat(api): add url to get result of a given message_id.

### PR DESCRIPTION
**Pull request by xdanielsb**

### CODE REVIEW

File: remoulade/api/main.py [hunk 1]
```diff
@@ -63,6 +65,23 @@ def cancel_message(message_id):
...
+@app.route("/messages/result/<message_id>")
+def get_results(message_id):
+    from ..message import get_encoder
+
+    try:
+        backend = remoulade.get_broker().get_result_backend()
+        result = backend.get_result(message_id)
+        if isinstance(result, BackendResult):
+            result = result.asdict()
+        encode_result = get_encoder().encode(result).decode("utf-8")  # type: ignore
```
Comment (antoinerabany) [inline remoulade/api/main.py:77]: catch exeption here (if encoded cannot be decoded for pickle for example).

File: remoulade/api/main.py [hunk 2]
```diff
@@ -63,6 +65,23 @@ def cancel_message(message_id):
...
+        if sys.getsizeof(encode_result) >= 1e4:
```
Comment (antoinerabany) [inline remoulade/api/main.py:78]: encode_result = "result too big". Add a comment to say how much 1e4 is in Mo

File: remoulade/api/main.py [hunk 3]
```diff
@@ -9,7 +10,8 @@
```
Comment (xdanielsb) [inline remoulade/api/main.py:10]: update changelog

File: remoulade/api/main.py [hunk 4]
```diff
@@ -78,6 +80,27 @@ def requeue_message(message_id):
...
+def get_results(message_id):
+    from ..message import get_encoder
+
+    max_size = 1e4
+    try:
+        backend = remoulade.get_broker().get_result_backend()
+        result = backend.get_result(message_id)
+        if isinstance(result, BackendResult):
+            result = result.asdict()
+        encode_result = get_encoder().encode(result).decode("utf-8")  # type: ignore
```
Comment (antoinerabany) [inline remoulade/api/main.py:93]: encoded_result

File: remoulade/api/main.py [hunk 5]
```diff
@@ -78,6 +80,27 @@ def requeue_message(message_id):
...
 
 
+@app.route("/messages/result/<message_id>")
+def get_results(message_id):
+    from ..message import get_encoder
+
+    max_size = 1e4
+    try:
+        backend = remoulade.get_broker().get_result_backend()
+        result = backend.get_result(message_id)
```
Comment (antoinerabany) [inline remoulade/api/main.py:90]: i would use
`result = Result(message_id=message_id)`
It's the way to do on remoulade.

File: remoulade/api/main.py [hunk 6]
```diff
@@ -78,6 +80,27 @@ def requeue_message(message_id):
...
+        if isinstance(result, BackendResult):
+            result = result.asdict()
+        encode_result = get_encoder().encode(result).decode("utf-8")  # type: ignore
+        size_result = sys.getsizeof(encode_result)
+        if size_result >= max_size:
+            encode_result = "The result is too big {}M".format(size_result / 1e6)
+        return {"result": encode_result}
+    except NoResultBackend:
```
Comment (antoinerabany) [inline remoulade/api/main.py:98]: handle ResultMissing. Also you could add a little message.

File: tests/state/test_state_api.py [hunk 7]
```diff
@@ -277,3 +278,38 @@ def test_requeue_message_with_pipeline(self, stub_broker, do_work, api_client, s
         res = api_client.get("messages/requeue/id1")
         assert res.json == {"error": "requeue message in a pipeline not supported"}
         assert res.status_code == 400
+
+    def test_get_result_message(self, stub_broker, stub_worker, api_client, state_middleware, redis_result_backend):
+        stub_broker.add_middleware(Results(backend=redis_result_backend))
```
Comment (antoinerabany) [inline tests/state/test_state_api.py:283]: you could add a fixture result_middleware a bit like state_middleware don't you think ?

Comment (xdanielsb) [reply tests/state/test_state_api.py:283]: yes, perfect

File: remoulade/api/main.py [hunk 8]
```diff
@@ -78,6 +82,28 @@ def requeue_message(message_id):
...
 
 
+@app.route("/messages/result/<message_id>")
+def get_results(message_id):
+    from ..message import get_encoder
+
+    max_size = 1e4
+    try:
+        result = Result(message_id=message_id).get()
+        if isinstance(result, BackendResult):
```
Comment (antoinerabany) [inline remoulade/api/main.py:92]: result is not a BackendResult, take a look at .get() mehtod. you can remove both next lines

File: remoulade/api/main.py [hunk 9]
```diff
@@ -78,6 +82,28 @@ def requeue_message(message_id):
...
+            result = result.asdict()
+        encoded_result = get_encoder().encode(result).decode("utf-8")
+        size_result = sys.getsizeof(encoded_result)
+        if size_result >= max_size:
+            encoded_result = "The result is too big {}M".format(size_result / 1e6)
+        return {"result": encoded_result}
+    except ResultMissing:
+        return {"result": "result is missing"}
+    except NoResultBackend:
+        return {"result": ""}
```
Comment (antoinerabany) [inline remoulade/api/main.py:102]: add a message for both next errors also.

**Pull request merged**